### PR TITLE
Ignore all config other than direct linked ID  from host page.

### PR DESCRIPTION
**Pull request by sheetaluk**
Fixes https://github.com/hypothesis/product-backlog/issues/209

### CODE REVIEW

**Comment by seanh** [comment]
@sheetaluk Can you fix the merge conflict? It's probably due to this merged PR: <https://github.com/hypothesis/client/pull/411>, `annotationQuery.extractAnnotationQuery()` is just `extractAnnotationQuery()` now

File: src/annotator/config.js [hunk 1]
```diff
@@ -16,6 +16,25 @@ function configFrom(window_) {
...
+  // In environments where the config has not been injected into the DOM,
+  // we try to retrieve it from the URL here.
+  var directLinkedID = annotationQuery.extractAnnotationQuery(window_.location.href);
+  if (directLinkedID) {
+    Object.assign(config, directLinkedID);
+  }
+
+  // Browser extensions should ignore all configuration from host page except
+  // annotationIDs
+  if (config.app.startsWith("chrome-extension://")) {
```
Comment (robertknight) [inline src/annotator/config.js]: In addition to "chrome-extension", we should also check for the "moz-extension" scheme used by Firefox Web Extensions. For Edge extensions, the scheme is "ms-browser-extension". See https://docs.microsoft.com/en-us/microsoft-edge/extensions/api-support/supported-apis

**Comment by sheetaluk** [comment]
1. Rebased
2. Made the changes to work with moz and edge
3. Wrote tests

Could you please review this once again @robertknight and @seanh ?

**Comment by seanh** [comment]
Looks good, I'm just going to give this a test locally then get it merged

**Comment by seanh** [comment]
Tested: the embedded client still reads the host page config on the publisher test site; the Chrome extension no longer reads config from the host page; direct links still work :+1:

**Pull request merged**